src/CodeStyle/CSharp/Analyzers/Formatting/Engine/CSharpFormatEngine.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Formatting.Rules;
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.CSharp.Formatting
+{
+    internal class CSharpFormatEngine : AbstractFormatEngine
```

Comment:
@dotnet/roslyn-ide Several additional files can be shared in this pull request if we can remove the use of `TaskExecutor` in the features layer and make format document operations single-threaded across the board.

Reply:
is something in TaskExecutor prevent something? running formatting concurrently in a big file for format document does help throughput. but not sure whether how many times that actually used in the wild. we might want to dig in telemetry data.

Reply:
The big throughput problem we're going to be hitting is Fix All in Solution. I'm interested to see cases where formatting performance has observable overhead in single-file situations (I've yet to see one of significant impact). Elimination of TaskExecutor provides a reduction in overall allocations, but I haven't measured the exact numbers yet.

Reply:
I went ahead and removed `TaskExecutor` from the formatter to substantially simplify the code.